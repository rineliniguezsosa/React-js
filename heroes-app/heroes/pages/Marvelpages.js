

export const Marvelpages = () => {
  return (
    <div>Marvelpages</div>
  )
}
