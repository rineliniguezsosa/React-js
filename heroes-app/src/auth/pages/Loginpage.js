import { useContext } from 'react'
import { useNavigate } from 'react-router-dom'
import { AuthContext } from '../context'


export const Loginpage = () => {
  const navigate = useNavigate()
  const onLogin = () =>{
    navigate('/',{
      replace:true
    })
  }
  return (
    <div className="container mt-5">
      <h1>Login</h1>
      <hr></hr>
      <button className="btn btn-primary" onClick={onLogin}>
        Login
      </button>
    </div>
  )
}
