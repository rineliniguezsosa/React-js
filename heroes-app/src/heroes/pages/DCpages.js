import { HeroList } from "../components/HeroList"


export const DCpages = () => {
  return (
    <>
    <h1>DC comics</h1>
    <hr></hr>
    </>
  )
}
