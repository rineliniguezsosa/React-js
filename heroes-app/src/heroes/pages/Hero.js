import { useParams } from "react-router-dom"


export const Hero = () => {
  return (
    <div>Hero</div>
  )
}
