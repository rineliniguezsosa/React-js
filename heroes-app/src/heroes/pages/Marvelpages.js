import { HeroList } from "../components/HeroList"


export const Marvelpages = () => {
  return (
    <div>Marvelpages</div>
  )
}
