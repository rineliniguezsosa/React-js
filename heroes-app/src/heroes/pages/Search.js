import { useNavigate } from "react-router-dom"
import { useForm } from "../hooks/useForm"

export const Search = () => {
  const {search,onInputChange} = useForm({search:''})

  const Sendform = (event)=>{
    event.preventDefault()
    if(search.trim().length <= 1) return;
    console.log(search)
  }
  return (
    <>
    <span>Search</span>
    <hr></hr>
    <div className="row">
    <div className="col-5">
      <h4>Busqueda.</h4>
      <hr></hr>
      <form onSubmit={Sendform}>
        <input 
        type="text" 
        placeholder="busqueda" 
        name="search"
        autoComplete="off"
        className="form-control"
        onChange={onInputChange}
        value={search}
        >
        </input>

        <button className="btn btn-outline-primary">Enviar</button>
      </form>
    </div>

    <div className="col-7">
      <h4>Resultados..</h4>
      <hr></hr>
      <div className="alert alert-primary">
          Busca un heroe
      </div>

      <div className="alert alert-danger">
          no hay resultados :/
      </div>
    </div>
    </div>
    </>
  )
}
