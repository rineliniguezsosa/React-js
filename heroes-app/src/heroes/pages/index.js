export * from './DCpages'
export * from './Marvelpages'