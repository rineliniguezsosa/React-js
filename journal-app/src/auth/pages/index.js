export * from './Loginpage'
export * from './Registropage'