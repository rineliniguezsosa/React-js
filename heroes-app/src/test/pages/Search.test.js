import {render,screen} from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom';
import { Search } from '../../heroes/pages/Search';



describe('prueba en Search.js', () => {
    test('debe de coincidir con el snapshot', () => {
        const { container } = render(
        <MemoryRouter><Search></Search></MemoryRouter>
        )

        expect(container).toMatchSnapshot();
    });
});