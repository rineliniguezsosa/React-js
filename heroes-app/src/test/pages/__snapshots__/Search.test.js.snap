// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba en Search.js debe de coincidir con el snapshot 1`] = `
<div>
  <span>
    Search
  </span>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Busqueda.
      </h4>
      <hr />
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="search"
          placeholder="busqueda"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary"
        >
          Enviar
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Resultados..
      </h4>
      <hr />
      <div
        class="alert alert-primary"
      >
        Busca un heroe
      </div>
      <div
        aria-label="danger-alert"
        class="alert alert-danger"
        style="display: none;"
      >
        no hay resultados con 
         :/
      </div>
    </div>
  </div>
</div>
`;
